extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/HybridJacksonPool.java

```diff
@@ -0,0 +1,175 @@
+package io.quarkus.jackson.runtime;
+
+import com.fasterxml.jackson.core.util.BufferRecycler;
+import com.fasterxml.jackson.core.util.BufferRecyclerPool;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+import java.util.function.Predicate;
+
+public class HybridJacksonPool implements BufferRecyclerPool {
```

Comment:
When this finalizes, we certainly want to have some Javadoc explaining the principle

Reply:
Done.